test/unit/domain_test.rb

```diff
@@ -20,6 +20,14 @@ def setup
     assert !other_domain.save
   end
 
+  test "should have a unique case-insensitive name" do
+    other_domain = Domain.new(:name => @domain.name.upcase)
+    refute other_domain.save
+    assert_raises ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid do
```

Comment:
The failure is actually because `rake test` runs `rake db:test:prepare` which loads the test db schema from `schema.rb`. However, the SQL commands I created for sqlite and pgsql aren't preserved in `schema.rb` as they aren't adapter agnostic. 
If you run only this test after running migrations on the test db it passes. 
I'm actually a bit stumped as to what would be the best approach here:
- remove this test knowing that it works and the problem is in the way rake loads the test DB schema.
- remove the case-insensitive enforcement from the DB altogether and rely on validations only
- create a duplicate column that saves the lowercase (similar to what we have for user names), causing this to work but introducing data duplication and increasing complexity for something the DB should do on its own.
- create something like foreigner that allows case-insensitive indices in `schema.rb` and migrations (most likely the most complex yet most reusable solution)

I'd be happy to get your input on what would be the best approach, as I will need to do the same for the next commit addressing the host names as well.